lua/entities/gmod_wire_expression2/core/find.lua

```diff
@@ -442,6 +442,29 @@ e2function entity findPlayerByName(string name)
 	return findPlayer(name)
 end
 
+--- Returns the player with the given SteamID
+e2function entity findPlayerBySteamID(string id)
+	if query_blocked(self, 1) then return NULL end
+	return player.GetBySteamID(id) or NULL
+end
+
+--- Returns the player with the given SteamID64
+e2function entity findPlayerBySteamID64(string id)
+	if query_blocked(self, 1) then return NULL end
+	return player.GetBySteamID64(id) or NULL
+end
+
+--- Returns the player with the given UserID
+e2function entity findPlayerByUserID(number id)
+	if query_blocked(self, 1) then return NULL end
+	for k, v in ipairs(player.GetAll()) do
```

Comment:
regular for loops should be faster than either ipairs or pairs. Pairs should be faster than ipairs because they both work the same way, but ipairs does additional checks to ensure that it's only iterating numerical sequential number keys.